### PR TITLE
Project app: Subject set picker

### PR DESCRIPTION
**Pull request by eatyourgreens**
Add a client-side subject set picker to the workflow selector links, which is shown on click when `workflow.grouped` is set (ie. when Panoptes expects you to pass a subject set ID with subject queries for that workflow.)

Screenshot of the picker, showing 3 subject sets from Shaun's test project, with subject images from Transformers and Overwatch.

TODO:
- [x] query Panoptes for the details of each subject set (display name, subject preview, completeness etc.)
- [x] style as per [the Invision design](https://projects.invisionapp.com/share/9WNF9M73AEN#/screens/423863706).
- [x] add tests.
- [x] write stories for the new components.

Package:
app-project

# Review Checklist

## General

- [x] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [x] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [x] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [x] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [x] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Package: `app-project`

This PR adds the "Subject Set Picker" component to the Project page.

- On a Project page with multiple workflows (e.g. `http://localhost:3000/projects/darkeshard/engaging-the-crowds-2020?env=production`), the "click on a workflow to begin the workflow" interaction has changed.
  - Clicking on a workflow - for a workflow that has `workflow.grouped=false` - will result in the user beginning the workflow as expected.
  - Clicking on a workflow - for a workflow that has `workflow.grouped=true` - will instead open the Subject Set Picker modal.
- The Subject Set Picker modal will list all Subject Sets belonging to the workflow ; clicking on a workflow will result in the user beginning the workflow, with the Subject Set selected via URL. (e.g. user will be directed to `http://localhost:3000/projects/darkeshard/engaging-the-crowds-2020/classify/workflow/16106/subject-set/85771`)  
- Subject Set Picker works as a standard modal ; it can be closed by clicking the X close button or pressing the Esc key. Tab navigation works fine (though see UI quirk)

UI quirks:
- (Known) The "close" button on the modal is difficult to see
- (Known) The height of each Subject Card is inconsistent
- (New?) when navigating via keyboard, the highlighted Subject Card has a very hard-to-notice focus style. Like, there's only a small highlighted square in the corner?

Screen Shot 2020-09-03 at 22 16 03

_Believe or not, the "Transformers series" has keyboard focus._

Most of the UI issues however have already been noted on Slack on the `#engaging-crowds` channel, so I'm not too concerned with flagging them.

Tested OK on the Engaging Crowds 2020 project mentioned earlier, on mac OS 10 + Chrome. Subject Set Picker works as expected for grouped workflows, and standard click-to-begin-workflow interaction for un-grouped workflows continue to function fine. 👍 

### Status

LGTM! 👍 

I didn't dig into the tests too deeply though - everything seems to pass, so I'm not sure if your earlier comments (on Slack, I think?) about 'tests being unfinished for this PR' is still accurate.

**Comment by eatyourgreens** [comment]
Thanks for taking a look @shaunanoordin. Re. Tests: I haven't written any tests for the picker itself, yet. 

The focus behaviour is odd. I see something similar in Firefox.

I've opened #1785 for the close button.

**Comment by eatyourgreens** [comment]
I haven’t written any stories for the subject set picker either.

File: packages/app-project/src/screens/ProjectHomePage/components/Hero/components/WorkflowSelector/components/SubjectSetPicker/components/SubjectSetCard/SubjectSetCard.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          src={subjectURLs[0]}
+          width={700}
+        />
+      </Box>
+      <Box
+        pad='small'
+      >
+        <SpacedText>{display_name}</SpacedText>
+        <Paragraph>
+          {set_member_subjects_count} subjects
```
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply packages/app-project/src/screens/ProjectHomePage/components/Hero/components/WorkflowSelector/components/SubjectSetPicker/components/SubjectSetCard/SubjectSetCard.js:58]: That's a good idea.

**Comment by eatyourgreens** [comment]
Here's a screenshot with the Annotate subject sets. Layout still needs a bit of work to match the designs.

Screenshot of Annotate,showing the subject set picker when there are enough subject sets to overflow the initial box.

**Review by beckyrother — COMMENTED** [review]
"the title bar of the modal vanishes off the top of the viewport when there are two or more rows of subject sets." 

Not sure what is meant here, would you mind posting a screenshot?

**Comment by eatyourgreens** [comment]
@beckyrother I thought the teal title bar had been cropped in the screenshot that I posted. Now that I look at it again, I'm not so sure. I'll work on a story for the storybook, so that we can check it with dummy content.

**Comment by eatyourgreens** [comment]
@beckyrother here it is in the storybook. The title styling should be fixed by #1787 

Screenshot of the light mode subject set picker in the storybook.

**Review by beckyrother — CHANGES_REQUESTED** [review]
Couple small spacing changes (sorry if some have been addressed by #1786 ): 

92924737-54b5d900-f431-11ea-9af3-5004efb01ea7

**Comment by eatyourgreens** [comment]
@beckyrother no problem. I think the spacing around the modal's content should be fixed by #1786. I've overridden the line-height on the heading, and the default margins on paragraphs, to set up the 15px spacing. Here's a screenshot with 30px/15px spacing. One question: how should the content of the grey box respond at small widths? Should the box padding remain at 30px and then the subject set cards be justified to fit?

Screenshot of the AnnoTate subject sets in a narrow box, so that three subject sets are visible and content overflows to the right.

**Comment by eatyourgreens** [comment]
Here's a 5 column grid on wide screens, 3 column grid on narrow screens.

Screenshot with 5 subject set cards per row and the picker covering the full width of the window.

Screenshot with 3 subject set cards per row and the picker modal not quite covering the window.

**Comment by beckyrother** [comment]
@eatyourgreens yep! That looks perfect, thank you!

**Comment by eatyourgreens** [comment]
Now that #1786 has merged, all modals have 20px spacing between the title bar and the content. I'm going to add an extra 10px margin to the top of the content for this one.

**Comment by eatyourgreens** [comment]
@beckyrother here it is with the changes from #1786. The close button colour and title bar height have been fixed, and I've added extra padding to the content at the top.

Subject set picker layout on a wide screen, with a purple layout grid overlaid.

Subject set picker layout on a narrow screen, with a purple layout grid overlaid.

**Comment by beckyrother** [comment]
Looks great, thanks @eatyourgreens !

**Pull request merged**